From 324d80be3f883229ce414dfec1c32fa64e7338c4 Mon Sep 17 00:00:00 2001
Message-ID: <324d80be3f883229ce414dfec1c32fa64e7338c4.1693142992.git.lorenzo@kernel.org>
From: Lorenzo Bianconi <lorenzo@kernel.org>
Date: Sun, 27 Aug 2023 15:26:32 +0200
Subject: [PATCH net-next 1/2] net: ethernet: mtk_wed: do not assume offload
 callbacks are always set

Check if wlan.offload_enable and wlan.offload_disable callbacks are set
since MT7996 SoC does not rely on them

Signed-off-by: Lorenzo Bianconi <lorenzo@kernel.org>
---
 drivers/net/ethernet/mediatek/mtk_wed.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/net/ethernet/mediatek/mtk_wed.c b/drivers/net/ethernet/mediatek/mtk_wed.c
index afc707729cf8..2cf615a0cb1f 100644
--- a/drivers/net/ethernet/mediatek/mtk_wed.c
+++ b/drivers/net/ethernet/mediatek/mtk_wed.c
@@ -1739,6 +1739,9 @@ int mtk_wed_flow_add(int index)
 	if (!hw || !hw->wed_dev)
 		return -ENODEV;
 
+	if (!hw->wed_dev->wlan.offload_enable)
+		return 0;
+
 	if (hw->num_flows) {
 		hw->num_flows++;
 		return 0;
@@ -1768,6 +1771,9 @@ void mtk_wed_flow_remove(int index)
 	if (!hw)
 		return;
 
+	if (!hw->wed_dev->wlan.offload_disable)
+		return;
+
 	if (--hw->num_flows)
 		return;
 
-- 
2.41.0

